src/mainframe/endpoints/report.py

```diff
@@ -218,7 +219,14 @@ def report_package(
             additional_information=body.additional_information,
         )
 
-        httpx.post(f"{mainframe_settings.reporter_url}/report/email", json=jsonable_encoder(report))
+        response = http_client.post(f"{mainframe_settings.reporter_url}/report/email", json=jsonable_encoder(report))
+        try:
+            response.raise_for_status()
+        except httpx.HTTPStatusError as err:
```

Comment:
Let's extract the logic for error checking and sending the request to a new function so it's easier to test.